examples/hello-world-pyprojecttoml/Cargo.toml

```diff
@@ -0,0 +1,23 @@
+[package]
+name = "hello-world"
+version = "0.1.0"
+edition = "2018"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+pyo3 = { version = "0.19.2", features = ["extension-module"] }
+
+[profile.release-lto]
+inherits = "release"
+lto = true
+
+[lib]
+# See https://github.com/PyO3/pyo3 for details
+name = "_lib"  # private module to be nested into Python package
+crate-type = ["cdylib"]
+path = "rust/lib.rs"
```

Comment:
Personally I would prefer if we left the rust directory as `src`, because that's what the rest of the examples do. I think there's no good reason to diverge here, but I'm also happy to be persuaded that `rust` is better. If `rust` is agreed to be better, we should update the rest of the examples to use it too.

Reply:
(See also https://github.com/PyO3/setuptools-rust/pull/350 - I've proposed moving to use `python/` subdirectory in all examples.)

Reply:
Hi @davidhewitt, thank you very much for the review.

As a very personal approach, I would avoid as much as I can to re-use the `src` directory for storing the Rust files. The reason behind this is that both Python/setuptools and Rust kind of compete for the convention of using the `src` directory. For example, when the user does not specify a `packages = [...]` configuration in `pyproject.toml`, `setup.cfg` and `setup.py`, setuptools might attempt to perform [automatic discovering](https://setuptools.pypa.io/en/latest/userguide/package_discovery.html), which will prioritize `src` as the convention for holding Python files meant for distribution.

Please note that the convention for using the `src` folder in Python is very old and well known in the community (e.g. [this blog post](https://blog.ionelmc.ro/2014/05/25/python-packaging/)). It even figures out in the [Packaging tutorial](https://packaging.python.org/en/latest/tutorials/packaging-projects/) and [Python packaging website](https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/).

Since both languages and tools (setuptools and Cargo) have a convention for using the `src` folder, I preferred to side step the discussion and be explicit when specifying path.

From the point of view of a Python user that does not have experience with Rust development, I believe that an example showing things explicit is very useful and helps to reduce the learning overhead.

If that is OK with you, I would refrain for modifying this PR in that direction (since is a direction that I have personal problems with), but if you or other maintainers would like to go ahead and implement modifications in a follow up PR, I completely respect the project's motivations for doing so.

Reply:
Fair enough, let's keep `rust` directory here 👍

Reply:
Thank you very much ❤️ 💚 💙 💜